pkg/cmd/target/set.go

```diff
@@ -169,3 +192,129 @@ var TargetSetCmd = &cobra.Command{
 		return nil
 	},
 }
+
+func handleTargetJSON(data []byte) error {
+	ctx := context.Background()
+
+	apiClient, err := apiclient_util.GetApiClient(nil)
+	if err != nil {
+		return err
+	}
+	var selectedTarget *target_view.TargetView
+	err = parseJSON(data, &selectedTarget)
+	if err != nil {
+		return fmt.Errorf("failed to parse input: %w", err)
+	}
+
+	if selectedTarget.Name == "" {
+		return errors.New("invalid input: 'name' field is required")
+	}
+	if selectedTarget.Options == "" {
+		return errors.New("option fields are required to setup your target")
+	}
+	targetManifest, res, err := apiClient.ProviderAPI.GetTargetManifest(ctx, selectedTarget.ProviderInfo.Name).Execute()
+	if err != nil {
+		return apiclient_util.HandleErrorResponse(res, err)
+	}
+	err = validateProperty(*targetManifest, selectedTarget)
+	if err != nil {
+		return err
+	}
+	targetData := apiclient.CreateProviderTargetDTO{
+		Name:    selectedTarget.Name,
+		Options: selectedTarget.Options,
+		ProviderInfo: apiclient.ProviderProviderInfo{
+			Name:    selectedTarget.ProviderInfo.Name,
+			Version: selectedTarget.ProviderInfo.Version,
+		},
+	}
+	res, err = apiClient.TargetAPI.SetTarget(ctx).Target(targetData).Execute()
+	if err != nil {
+		return apiclient_util.HandleErrorResponse(res, err)
+	}
+	views.RenderInfoMessage("Target set successfully and will be used by default")
+	return nil
+}
+
+func parseJSON(data []byte, v interface{}) error {
+	if err := json.Unmarshal(data, v); err == nil {
+		return nil
+	}
+	return errors.New("input is not a valid JSON")
+}
+
+func validateProperty(targetManifest map[string]apiclient.ProviderProviderTargetProperty, target *target_view.TargetView) error {
+	optionMap := make(map[string]interface{})
+	if err := json.Unmarshal([]byte(target.Options), &optionMap); err != nil {
+		return fmt.Errorf("failed to parse options JSON: %w", err)
+	}
+	for optionKey := range optionMap {
+		if _, exists := targetManifest[optionKey]; !exists {
+			return fmt.Errorf("invalid property '%s' for target Manifest '%s'", optionKey, target.Name)
+		}
+	}
+
+	sortedKeys := make([]string, 0, len(targetManifest))
+	for k := range targetManifest {
+		sortedKeys = append(sortedKeys, k)
+	}
+	sort.Strings(sortedKeys)
+
+	for _, name := range sortedKeys {
```

Comment:
Keys in the target options should be considered optional.

E.g. it's valid to set a target in aws with the Access Key Id not set in the target options.

So if the user doesn't set all properties defined in the manifest, it shouldn't throw an error. We can implement required checks in the manifest later

Reply:
Yes, i have added it [here](https://github.com/daytonaio/daytona/pull/1380/files#diff-6559a0204339e2a60f691c723ba5914cd2edac64ae210296aac6c40d38338215R264), it skips  validation if the property is not set in optionMap, making it optional